pkg/factsengine/entities/fact_value.go

```diff
@@ -28,13 +41,13 @@ type FactValue interface {
 	AsInterface() interface{}
 }
 
-// NewFactValue constructs a FactValue from a nested interface.
-func NewFactValue(factInterface interface{}) (FactValue, error) {
+// NewFactValueWithConf constructs a FactValue from a nested interface with a provided configuration
+func NewFactValueWithConf(factInterface interface{}, conf *Conf) (FactValue, error) {
```

Comment:
After the renaming of the proxy constructor we can safely revert that function to `NewFactValue`